core/arch/arm/kernel/thread_spmc.c

```diff
@@ -326,46 +327,104 @@ static bool is_my_uuid(uint32_t w0, uint32_t w1, uint32_t w2, uint32_t w3)
 	       w2 == 0x02005ebc && w3 == 0x1bc5d5a5;
 }
 
-static void handle_partition_info_get(struct thread_smc_args *args,
-				      struct ffa_rxtx *rxtx)
+void spmc_fill_partition_entry(struct ffa_partition_info *fpi,
+			       uint16_t endpoint_id)
 {
-	uint32_t ret_fid = 0;
-	int rc = 0;
+	fpi->id = endpoint_id;
+	/* Number of execution contexts implemented by this partition */
+	fpi->execution_context = 1;
 
-	if (!is_nil_uuid(args->a1, args->a2, args->a3, args->a4) &&
-	    !is_my_uuid(args->a1, args->a2, args->a3, args->a4)) {
-		ret_fid = FFA_ERROR;
-		rc = FFA_INVALID_PARAMETERS;
-		goto out;
+	/*
+	 * BIT(0): Supports receipt of direct requests
+	 * BIT(1): Can send direct requests
+	 * BIT(2): Cannot send and receive indirect messages
+	 * BIT(3): Does not support receipt of notifications
+	 * BIT(4-5): Partition ID is a PE endpoint ID
+	 */
+	fpi->partition_properties = BIT(0) | BIT(1);
+}
+
+static uint32_t handle_partition_info_get_all(int *rc, struct ffa_rxtx *rxtx)
+{
+	struct ffa_partition_info *fpi = NULL;
+
+	fpi = rxtx->tx;
+
+	/* Add OP-TEE SP */
+	spmc_fill_partition_entry(fpi, my_endpoint_id);
+	rxtx->tx_is_mine = false;
+	*rc = 1;
+	fpi++;
+
+	if (IS_ENABLED(CFG_SECURE_PARTITION)) {
+		size_t elements = (rxtx->size / sizeof(*fpi)) - 1;
+
+		if (sp_get_all_partitions(fpi, &elements))
+			return TEE_ERROR_OUT_OF_MEMORY;
```

Comment:
i think this is a wrong solution.
caller would think the TEE lacks RAM. Prefer restrict TEE_ERROR_OUT_OF_MEMORY to its real purpose.
Here returning TEE_ERROR_BUFFER_TOO_SMALL should be preferred.

Reply:
Done